README.md

```diff
@@ -1,25 +1,87 @@
+$ composer require silverstripe/security-extensions 1.x-dev
 ```
 
-## Documentation
+## Features
 
-Developer and user guide documentation [is available in the "docs" directory](docs/en/index.md).
+### Sudo mode
+
+Sudo mode represents a heightened level of permission in that you are more certain that the current user is actually
+the person whose account is logged in. This is performed by re-validating that the account's password is correct, and
+will then last for a certain amount of time (configurable) until it will be checked again.
+
+#### Enabling sudo mode for controllers
+
+You can add the `SilverStripe\SecurityExtensions\Services\SudoModeServiceInterface` as a dependency to a controller
+that requires sudo mode for one of its actions:
+
+```php
+class MyController extends Controller
+{
+    private $sudoModeService;
+    
+    private static $dependencies = ['SudoModeService' => '%$' . SudoModeServiceInterface::class];
+    
+    public function setSudoModeService(SudoModeServiceInterface $sudoModeService): self
+    {
+        $this->sudoModeService = $sudoModeService;
+        return $this;
+    }
+}
+```
+
+Performing a sudo mode verification check in a controller action is simply using the service to validate the request:
+
+```php
+public function myAction(HTTPRequest $request): HTTPResponse
+{
+    if (!$this->sudoModeService->check($request->getSession()) {
+        return $this->httpError(403, 'Sudo mode is required for this action');
+    }
+    // ... continue with sensitive operations
+}
+``` 
+
+### Using sudo mode in a React component
+
+This module defines a [React Higher-Order-Component](https://reactjs.org/docs/higher-order-components.html) which can
+be applied to React components in your module or code to intercept component rendering and show a "sudo mode required"
+information and login screen, which will validate, activate sudo mode, and re-render the wrapped component afterwards
+on success.
+
+**Note:** the JavaScript injector [does not currently support injecting transformations/HOCs](https://github.com/silverstripe/react-injector/issues/4),
+so we have coupled the application of these [injector transformations](https://docs.silverstripe.org/en/4/developer_guides/customising_the_admin_interface/reactjs_redux_and_graphql/#transforming-services-using-middleware)
+into this module itself for the silverstripe/mfa module. Unfortunately, if you want to apply this to your own code
+you will need to either duplicate the `SudoMode` HOC into your project or module and apply the transformation at that
+point.
+
+![Sudo mode HOC example](docs/_images/sudomode.png)
+
+#### Requirements for adding to a component
+
+You will need to ensure that the component's respective PHP FormField has the following attributes added to its
+form schema (see `FormField::getSchemaData()`):
+
+* `SecurityID`: for Cross-Site-Request-Forgery protection - can be obtained via `SecurityToken::inst()->getValue()`
```

Comment:
@ScopeyNZ you were saying you want to use lib/Config for this right?

Reply:
Yeah on client side we can just: 
```
import Config from 'lib/Config';

const token = Config.get('SecurityID');
```